### DISCUSSION TITLE
Please, provides support for Arduino and AVR with coc-extension

### DISCUSSION THREAD

**Discussion by MuhammadMuradG** [neoclide/coc.nvim#2238] [Extension request]
**Is your feature request related to a problem? Please describe.**
I tried to use `coc-clangd` but it was complicated to adjust and also when I tried to compile the project. I think it misses a lot things to be acceptable for using as a development tool for AVR (Arduino is a product that uses an AVR microcontrollers and the name of the product varies according to the type of used AVR microcontroller. However, Arduino has a lot of package that may be simplified the implementation of extension).

**Describe the solution you'd like**
I suggest `coc-avr` or at least `coc-arduino` extension for support AVR programming which include Arduino support. There is a VS Code plugin for arduino [vscode-arduino](https://marketplace.visualstudio.com/items?itemName=vsciot-vscode.vscode-arduino)

**Comment by phaalonso** [comment]
I need it too

**Comment by Plunne** [comment]
Same, i need it too

**Comment by sandikodev** [comment]
yea... with espressif/esp support

**Reply by purofle** [reply]
I think you can use the ccls and platformio

**Comment by Charl13** [comment]
If you end up here searching for a way to make Vim work with Arduino via CoC then using the official [Arduino Language Server](https://github.com/arduino/arduino-language-server) of the [Arduino IDE v2](https://github.com/arduino/arduino-ide) (still in beta at time of writing) is the way to go.

There is an [example snippet](https://github.com/neoclide/coc.nvim/wiki/Language-servers#Arduino) of the coc-settings.json to make this work.

**Comment by K4LCIFER** [comment]
I would love this as well.
